src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/SingleInitNullable.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using Roslyn.Utilities;
+
+/// <summary>
+/// A lazily initialized version of <see cref="Nullable{T}"/> which uses the same space as a <see cref="Nullable{T}"/>.
+/// </summary>
+[NonCopyable]
+internal struct SingleInitNullable<T>
```

Comment:
I admit the "Nullable" name here doesn't really fit since you don't use this like a regular nullable much anymore, it's just a single init helper.

Reply:
yeah, ti's a bit odd.  i do like 'Nullabel' in that you could use it like Nullable to indicate it has a 'null' non-computed state (teh initial state), and then it transitions to the non-null state. it just does so safely.

Reply:
Yeah you never use the "null" though -- that's all internal. Just maybe SingleInitStruct?